cli/lsp/resolver.rs

```diff
@@ -181,8 +196,18 @@ impl LspResolver {
     Ok(())
   }
 
-  pub fn as_graph_resolver(&self) -> &dyn Resolver {
-    self.graph_resolver.as_ref()
+  pub fn as_graph_resolver<'a>(
+    &self,
+    // this really only needs a HashSet<ModuleSpecifier>, but it's provided
+    // the entire HashMap to avoid cloning all the time
+    open_docs: &'a HashMap<ModuleSpecifier, Arc<Document>>,
```

Comment:
This is only necessary for sloppy import resolution. It's slightly unfortunate to have to have to pass it around, but I'm not sure a better way because we're mutating the `Documents` while calling this sometimes and sloppy import resolution needs to know about files that may only exist in memory.